src/xpcc/architecture/platform/board/nucleo_f031k6/nucleo_f031k6.hpp

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2016, Roboterclub Aachen e.V.
+ * All Rights Reserved.
+ *
+ * The file is part of the xpcc library and is released under the 3-clause BSD
+ * license. See the file `LICENSE` for the full license governing this code.
+ * ------------------------------------------------------------------------ */
+
+//
+// NUCLEO-F103RB
+// Nucleo kit for STM32F103RB
+// http://www.st.com/web/catalog/tools/FM116/SC959/SS1532/LN1847/PF259875
+//
+
+#ifndef XPCC_STM32_NUCLEO_F031K6_HPP
+#define XPCC_STM32_NUCLEO_F031K6_HPP
+
+#include <xpcc/architecture/platform.hpp>
+#include <xpcc/debug/logger.hpp>
+#define XPCC_BOARD_HAS_LOGGER
+
+using namespace xpcc::stm32;
+
+
+namespace Board
+{
+
+/// STM32F031K6 running at 48MHz generated from the internal 8MHz crystal
+// Dummy clock for devices
+struct systemClock {
+	static constexpr uint32_t Frequency = MHz48;
+	static constexpr uint32_t Ahb = Frequency;
+	static constexpr uint32_t Apb = Frequency;
+
+	static constexpr uint32_t Adc1   = Apb;
+
+	static constexpr uint32_t Spi1   = Apb;
+
+	static constexpr uint32_t Usart1 = Apb;
+
+	static constexpr uint32_t I2c1   = Apb;
+
+	static constexpr uint32_t ApbTimer = Apb * 1;
+	static constexpr uint32_t Timer1  = ApbTimer;
+	static constexpr uint32_t Timer2  = ApbTimer;
+	static constexpr uint32_t Timer3  = ApbTimer;
+	static constexpr uint32_t Timer4  = ApbTimer;
+	static constexpr uint32_t Timer5  = ApbTimer;
+	static constexpr uint32_t Timer6  = ApbTimer;
+
+	static bool inline
+	enable()
+	{
+		ClockControl::enableInternalClock();	// 8MHz
+		// internal clock / 2 * 16 = 64MHz, => 64/1.5 = 42.6 => bad for USB
+		ClockControl::enablePll(ClockControl::PllSource::InternalClock, 12, 1);
+		// set flash latency for 64MHz
+		ClockControl::setFlashLatency(Frequency);
+		// switch system clock to PLL output
+		ClockControl::enableSystemClock(ClockControl::SystemClockSource::Pll);
+		ClockControl::setAhbPrescaler(ClockControl::AhbPrescaler::Div1);
+		// APB1 has max. 36MHz
+		ClockControl::setApbPrescaler(ClockControl::ApbPrescaler::Div1);
+		// update frequencies for busy-wait delay functions
+		xpcc::clock::fcpu     = Frequency;
+		xpcc::clock::fcpu_kHz = Frequency / 1000;
+		xpcc::clock::fcpu_MHz = Frequency / 1000000;
+		xpcc::clock::ns_per_loop = ::round(3000 / (Frequency / 1000000));
+
+		return true;
+	}
+};
+
+// Arduino Nano Footprint
+using A0 = GpioA0;
+using A1 = GpioA1;
+using A2 = GpioA3;
+using A3 = GpioA4;
+using A4 = GpioA5;
+using A5 = GpioA6;
+using A6 = GpioA7;
+using A7 = GpioA2;
+
+using D0  = GpioA10;
+using D1  = GpioA9;
+using D2  = GpioA12;
+using D3  = GpioB0;
+using D4  = GpioB7;
+using D5  = GpioB6;
+using D6  = GpioB1;
+using D7  = GpioF0;
+using D8  = GpioF1;
+using D9  = GpioA8;
+using D10 = GpioA11;
+using D11 = GpioB5;
+using D12 = GpioB4;
+using D13 = GpioB3;
+
+using Button = xpcc::GpioInverted<GpioInputA12>;
```

Comment:
The board I got from STM had a shunt across D2/GND, acting in the same capacity as the user button on the bigger boards. I wasn't aware of xpcc::GpioUnused though, so I will use that instead.